### PR TITLE
add gulp build pipeline to online demo

### PR DESCRIPTION
**Pull request by arlimus**
starting point, more to do 

**why does it matter?**
to build an offline version of the online demo.
we push it to gh-pages via our build process next

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @arlimus

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

File: www/tutorial/gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      'tutorial_files/**',
+      'node_modules/shellwords/src/shellwords.coffee',
+    ])
+    .pipe(gcopy('dist/'));
+})
+
+gulp.task('build', ['webpack', 'copy'], function(){
+  return gulp.src('index.html')
+    .pipe(useref())
+    .pipe(debug({ title: 'Uglify scripts' }))
```
Comment (chris-rock) [inline www/tutorial/gulpfile.js:33]: We may think about using the uglify webpack plugin

Comment (chris-rock) [inline www/tutorial/gulpfile.js:33]: https://webpack.github.io/docs/list-of-plugins.html#uglifyjsplugin

Comment (arlimus) [reply www/tutorial/gulpfile.js:33]: Great suggestion, will add uglify to webpack as well. We will need it in both places.

**Pull request merged**